pages/multiplayerGame.js

```diff
@@ -0,0 +1,159 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import config from '../src/config';
+import { Router } from '../src/routes';
+
+import Bloc from '../src/components/bloc';
+import Button from '../src/components/button';
+import Grid from '../src/components/grid';
+import Page from '../src/components/page';
+import Row from '../src/components/row';
+import Section from '../src/components/section';
+
+import { game, move } from '../src/services/multiplayerGameService';
+
+export default class MultiplayerGame extends Component {
+    state = {
+        isLoading: true,
+        id: -1,
+        otherPlayerId: -1,
+        isMultiplayer: false,
+        token: '',
+        currentGrid: [],
+        turn: -1,
+        winnerId: -1,
+    };
+
+    static propTypes = {
+        id: PropTypes.string.isRequired,
+        token: PropTypes.string.isRequired,
+    };
+
+    static getInitialProps = async ({ query }) => ({
+        id: query.id,
+        token: query.token,
+    });
+
+    waitForOtherPlayer = async (id, token) => {
+        const { otherPlayer } = await game()(id, token);
+        if (otherPlayer) {
+            return Promise.resolve(otherPlayer);
+        }
+
+        await new Promise(resolve => {
+            setTimeout(resolve, config.refreshDuration);
+        });
+
+        return this.waitForOtherPlayer(id, token);
+    };
+
+    requestGame = async (id, token) => {
```

Comment:
Would be a lot better (and easier to test) if this component was a plain dumb graphical component and the state manipulation was done in a HOC like this [recompose example](https://github.com/acdlite/recompose/blob/master/docs/API.md#withstate)

This is applicable for all of the components on this PR.

Reply:
I'll do this refactoring in another PR :)